src/dfp/render-advert.ts

```diff
@@ -90,6 +90,16 @@ sizeCallbacks[adSizes.outstreamMobile.toString()] = (advert: Advert) =>
 sizeCallbacks[adSizes.googleCard.toString()] = (advert: Advert) =>
 	advert.updateExtraSlotClasses('ad-slot--gc');
 
+sizeCallbacks[adSizes.billboard.toString()] = (advert: Advert) => {
+	if (
+		advert.node.dataset.name === 'merchandising' ||
+		advert.node.dataset.name === 'merchandising-high'
+	) {
+		advert.node.classList.add('ad-slot--merchandising-billboard');
+	}
+	return Promise.resolve();
```

Comment:
Is it appropriate to instead use advert.updateExtraNodeClasses here, as used in the cases above? I can’t find any examples of extraNodeClasses, so I’m not sure if this part of the advert API is still useful.

Reply:
Had to add `await` in refactored code when using the above function, as its asynchronous. I believe its because of the if is awaiting to decide on the value before conditionally adding the class.

Reply:
You should be able to just return the promise from `updateExtraSlotClasses`

```js
	if (
		advert.node.dataset.name === 'merchandising' ||
		advert.node.dataset.name === 'merchandising-high'
	) {
		return advert.updateExtraSlotClasses('ad-slot--merchandising-billboard');
	}
	return Promise.resolve();
```

Reply:
That was my initial solution, but wound up in all sorts of type issues?
```
Unsafe member access .updateExtraSlotClasses on an `any` value.eslint[@typescript-eslint/no-unsafe-member-access](https://typescript-eslint.io/rules/no-unsafe-member-access)
```